core/src/main/scala/org/apache/spark/sql/execution/CodegenSparkFallback.scala

```diff
@@ -87,6 +88,15 @@ case class CodegenSparkFallback(var child: SparkPlan) extends UnaryExecNode {
             }
           case None => throw t
         }
+    } finally {
+      // SNAP-1422
+      val cacheMetaDataForConnector =
```

Comment:
Reading property in every query run does not make sense. This can be made a object var.